ehrbase/openEHR_SDK
# Use sonar workflow_run action

# Changes

Use the extracted sonar analyses from this workflow_run -> https://github.com/ehrbase/openEHR_SDK/blob/develop/.github/workflows/report-sonar-results.yml.

The build logic is now:

#1 Build & Test
  - run unit tests with coverage
  - run integration tests with coverage and combine results
  - upload overall coverage
  
#2 Sonar Report 
  - download overall coverage
  - run sonar report with workflow_run.pullrequest target   

This allow to use the sonar report also for PR from forked repositories.


other changes:
- adjusted build and test name to be aligned with ehrbase
- adjusted codeql & checkstyle wotkflows definitions and name to be aligned with ehrbase


If this change is ok and merge I'll check it with a dependabot PR. In case this succeeds I will adopt the ehrbase repo sonar analyses in the same way.

# Pre-Merge checklist

- [x] New code is tested
- [x] Present and new tests pass
- [x] The build is working without errors
- [x] No new Sonar issues introduced
